Kitodo-API/src/main/java/org/kitodo/api/filemanagement/LockingResult.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.api.filemanagement;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Manages allocated locks or provides information about conflict originators in
+ * case of error.
+ */
+public interface LockingResult extends AutoCloseable {
```

Comment:
I think "LockingResult" is a missleading name. The Javadoc says:"manages" maybe "LockingManager" is more appropriate?

Reply:
Maybe something like `LockProtocol` since it's defining the commands you can pass on.

Reply:
I would not say so. OK, you could rename it to `LockProtocol`. But then you would have to rename more, because then the whole metaphor does not match any more, isn’t it? At the moment, you call `tryLock()`. As a result, you get a `LockResult`, and that offers, for example, the method `isSuccesful()`. The result of trying to lock is successful, or maybe not. If not, I can `getConflicts()`, the conflict is a possible _result_ of the attempt to lock. If I rename that to _protocol_ now—on my request, trying to lock a file, will I get back a _protocol_ that I can then ask if it succeeded? What is a successful protocol? And then I can ask the _protocol_ if it has a conflict? I find this inconsistent, there was a conflict while trying to lock, and not a conflict in the protocol, then would rather thrown an `Exception`, isn’t it? A conflict in the protocol would be more likely for me if I request a copy of a file but the file can not be copied, not because of a locking conflict with another user, but, e.g. because the disk is full...

But if you find that much easier to understand, I rename it.

Reply:
Okay, let's do this…!